docs/algorithms/mathematics-algorithms/euclidean-algorithm.md

```diff
@@ -0,0 +1,102 @@
+---
+id: euclidean-algorithm
+title: Euclidean Algorithm in Number Theory
+sidebar_label: Euclidean Algorithm
+sidebar_position: 1
+description: Explanation and implementation of the Euclidean Algorithm to find the GCD of two numbers.
+tags: [number-theory, gcd, algorithms]
+---
+
+# Euclidean Algorithm
+
+The Euclidean Algorithm is an efficient method for finding the Greatest Common Divisor (GCD) of two integers. It uses the principle that the GCD of two numbers does not change if the larger number is replaced by its remainder when divided by the smaller number.
+
+## Steps to Implement
+1. Divide the larger number by the smaller number and find the remainder.
+2. Replace the larger number with the smaller number and the smaller number with the remainder.
+3. Repeat until the remainder is 0. The non-zero remainder is the GCD.
+
+## Code Examples
+
+### C++ Implementation
+
+```cpp
+#include <iostream>
+using namespace std;
+
+int gcd(int a, int b) {
+    while (b != 0) {
+        int remainder = a % b;
+        a = b;
+        b = remainder;
+    }
+    return a;
+}
+
+int main() {
+    int a, b;
+    cout << "Enter two integers: ";
+    cin >> a >> b;
+    cout << "GCD of " << a << " and " << b << " is: " << gcd(a, b) << endl;
+    return 0;
+}
+```
+
+### Python Implementation
+
+```python
+def gcd(a, b):
+    while b != 0:
+        a, b = b, a % b
+    return a
+
+if __name__ == "__main__":
+    a = int(input("Enter the first integer: "))
+    b = int(input("Enter the second integer: "))
+    print(f"GCD of {a} and {b} is: {gcd(a, b)}")
+```
+
+## Example Walkthrough
+
+### Example 1: GCD of 56 and 98
+1. $$ 98 \mod 56 = 42 $$
+2. $$ 56 \mod 42 = 14 $$
+3. $$ 42 \mod 14 = 0 $$
+
```

Comment:
write in good format because

`$` is used for inline and `$$` is used for block and content in center so, follow best practices rule.